pkg/addonmanager/controllers/agentdeploy/controller.go

```diff
@@ -150,20 +150,21 @@ func (c *addonDeployController) sync(ctx context.Context, syncCtx factory.SyncCo
 	}
 
 	cluster, err := c.managedClusterLister.Get(clusterName)
-	if errors.IsNotFound(err) {
-		// to delete the addon to make sure the addon is cleaned up when the cluster is not found.
-		// need go to sync to handle cleanup if the addon is deleting.
-		// in the normal case the addon should have been cleaned up after the cluster is deleted.
-		if addon.DeletionTimestamp.IsZero() {
-			return c.cleanupAddon(ctx, clusterName, addonName)
-		}
+
+	// waiting for the addon is deleted after cluster is deleted.
+	// need go to sync to handle cleanup if the addon is deleting.
+	// The other controller should delete the addon once cluster is deleted.
+	// TODO: should consider to force delete the addon and its manifestWork because there may be no work-agent in spoke cluster.
+	if errors.IsNotFound(err) && addon.DeletionTimestamp.IsZero() {
+		return nil
 	}
 	if err != nil {
```

Comment:
if the err is not found err and addon is in deletion, will it return not found error here?

Reply:
yes.   fixed to return when meet err. aded a TODO here.